cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -248,6 +256,42 @@ def _get_data(self):
             )
             return False
 
+    def _crawl_metadata(self):
+        start_time = time.monotonic()
+
+        self.userdata_raw = ec2.get_instance_userdata(
+            self.api_ver, self.metadata_address
+        )
+        LOG.debug(self.userdata_raw)
+        self.metadata = ec2.get_instance_metadata(
+            self.api_ver, self.metadata_address
+        )
+        LOG.debug(self.metadata)
+        LOG.debug(
+            "Crawl of metadata service took %s seconds",
+            int(time.monotonic() - start_time),
+        )
```

Comment:
1. It's a security danger to log metadata as it could contain ssh private keys, passwords, URL credentials etc. This has resulted in a couple of CVEs in the past for cloud-init on various platforms, so let's not log(self.metadata) (user-data or vendor-data too).
2.  We should drop the start_time log calculation and just use our new performance. Timer decorator instead.
3. Let's make this method a public method on the class. A while back (f0bc02d7e2) we thought it'd be nice to standardize a public `crawl_metadata` method that is a read-only operation to walk the current IMDS config data in case we wanted to surface a command line tool to `get-latest-imds-data`. While that subcommand hasn't surfaced, there's nothing wrong with continuing to lay that groundwork in this PR.
```suggestion
    @performance.timed("Crawling metadata", log_mode="always")
    def _crawl_metadata(self):
        self.userdata_raw = ec2.get_instance_userdata(
            self.api_ver, self.metadata_address
        )
        self.metadata = ec2.get_instance_metadata(
            self.api_ver, self.metadata_address
        )
```

Reply:
Sounds good! The new decorator provides a nice syntax